# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.h
# Commit: 19dcca77a1cf
# Full Hash: 19dcca77a1cf1c590dfda293418aec84cce701de
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-24 09:31:07
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P1 allow OnStartRequest go thru pBg IPC, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73529
# ==============================================================================

diff -r 91c26c5b1e33 -r 19dcca77a1cf netwerk/protocol/http/HttpChannelChild.h
--- a/netwerk/protocol/http/HttpChannelChild.h	Tue Jun 23 18:34:42 2020 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.h	Tue Jun 23 16:56:15 2020 +0000
@@ -267,6 +267,10 @@
   already_AddRefed<nsIEventTarget> GetODATarget();
 
   [[nodiscard]] nsresult ContinueAsyncOpen();
+  void ProcessOnStartRequest(const nsHttpResponseHead& aResponseHead,
+                             const bool& aUseResponseHead,
+                             const nsHttpHeaderArray& aRequestHeaders,
+                             const HttpChannelOnStartRequestArgs& aArgs);
 
   // Callbacks while receiving OnTransportAndData/OnStopRequest/OnProgress/
   // OnStatus/FlushedForDiversion/DivertMessages on background IPC channel.